### PR TITLE
Adapt multisig metadata builder to 3 variants

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Supports 2 new multisig ISM variants from https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2173 in relayer.

### Drive-by changes

- migrates relayer keys to be anvil accounts
- moves from integer to enum module types

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2221

### Backward compatibility

Yes

### Testing

E2e tests

### CODE REVIEW

File: rust/agents/relayer/Cargo.toml [hunk 1]
```diff
@@ -33,6 +33,7 @@ hyperlane-base = { path = "../../hyperlane-base" }
 hyperlane-ethereum = { path = "../../chains/hyperlane-ethereum" }
 num-derive.workspace = true
 num-traits.workspace = true
+derive_more = "0.99.17"
```
Comment (mattiecnvr) [inline rust/agents/relayer/Cargo.toml:36]: If we specify the patch version it won't auto update for security/bug fixes.
```suggestion
derive_more = "0.99"
```

File: rust/chains/hyperlane-ethereum/src/interchain_security_module.rs [hunk 2]
```diff
@@ -96,9 +97,9 @@ where
     M: Middleware + 'static,
 {
     #[instrument(err, ret)]
-    async fn module_type(&self) -> ChainResult<u8> {
+    async fn module_type(&self) -> ChainResult<ModuleType> {
         let module_type = self.contract.module_type().call().await?;
-        Ok(module_type)
+        Ok(ModuleType::from_u8(module_type).unwrap_or_default())
```
Comment (mattiecnvr) [inline rust/chains/hyperlane-ethereum/src/interchain_security_module.rs:102]: Do we want to log if we encounter an unknown module type? (Maybe you already do elsewhere?)

**Review by mattiecnvr — APPROVED** [review]

File: rust/agents/relayer/src/msg/metadata/base.rs [hunk 3]
```diff
@@ -144,7 +134,29 @@ impl BaseMetadataBuilder {
...
+                validators,
+                threshold,
+                message.nonce,
+                highest_known_nonce,
+            )
+            .await
+            .context(CTX)
+    }
+
+    pub async fn fetch_checkpoint(
```
Comment (asaj) [inline rust/agents/relayer/src/msg/metadata/base.rs:147]: You need two different versions of this function for censorship resistance.

For merkle proof ISMs, you can use *any* checkpoint with nonce >= message nonce. You will need something similar to `fetch_checkpoint_in_range`

For message ID ISMs, you need the specific checkpoint that matches the message nonce, which is what you are doing here.

**Review by asaj — CHANGES_REQUESTED** [review]

**Comment by asaj** [comment]
Also I'd imagine there's a lot of code that can be saved by using generics?

File: rust/chains/hyperlane-ethereum/src/interchain_security_module.rs (hunk 2 above)
Comment (asaj) [reply rust/chains/hyperlane-ethereum/src/interchain_security_module.rs:102]: Yeah I think debug or maybe warn

**Pull request merged**